pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/ast/jdkversiontests/java16p/expression/ConstantExpr.java

```diff
@@ -1,9 +0,0 @@
-/*
- * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
- */
-package com.example.expression;
-
-/**
- * @see <a href="https://openjdk.java.net/jeps/397">JEP 397: Sealed Classes (Second Preview)</a>
- */
-public final class ConstantExpr implements Expr { }
```

Comment:
Why don't we keep these tests, if the features are identical? Where do the new files for java18p come?

Reply:
Maybe approach is this: delete the old preview tests (in that case 16-preview) without looking at it at all. If the preview features was standardized, then we should have a test for e.g. 17. In that case, these files are really in 17 already (https://github.com/pmd/pmd/blob/master/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/ast/jdkversiontests/java17/expression/ConstantExpr.java). 

When adding a new language preview version, I copy the previous test files. For this update, I copied 17-preview into 18-preview and reviewed the test cases. E.g. I added a bit more code samples from the new JEP. Luckily, there were no grammar changes. I didn't touch 17-preview at all. But 18-preview contains all cases from 17-preview plus some more.

So rule of thumb: whenever a preview feature is standardized, we need to take over the test cases from preview (maybe with additional changes).

Reply:
Hm... we might think about whether to keep 17-preview when we add 19 in September later this year. 17 is a LTS release, so potentially someone could need it. The only preview feature from 17-preview is pattern matching for switch, which would be standardized in 19 though (probably).